.github/workflows/e2e.yaml

```diff
@@ -115,3 +123,35 @@ jobs:
         name: diagnostics-e2e-tests
         path: /tmp/ktf-diag*
         if-no-files-found: ignore
+
+    - name: collect test report
+      uses: actions/upload-artifact@v3
+      with:
+        name: tests-report
+        path: istio-tests.xml
+
+  buildpulse-report:
+    environment: "Configure ci"
+    needs:
+      - "e2e-tests"
+      - "istio-tests"
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: download tests report
+        id: download-coverage
+        uses: actions/download-artifact@v3
+        with:
+          name: tests-report
+          path: report
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: ${{ !cancelled() }}
+        uses: Workshop64/buildpulse-action@main
+        with:
+          account: 962416
+          repository: 127765544
+          path: report/*.xml
+          key: ${{ secrets.BUILDPULSE_ACCESS_KEY_ID }}
+          secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
```

Comment:
Be sure to add those to 1pass

Reply:
Do you think it's gonna be needed even if the secrets are accessible for every team member via https://buildpulse.io/@Kong/kubernetes-ingress-controller/setup/github-actions?

Reply:
We'd have to document that somewhere so that folks know how to find it

Reply:
https://github.com/Kong/team-k8s/pull/253